packages/playground/website/public/puzzle.php

```diff
@@ -0,0 +1,202 @@
+            The topic should be fun and not too serious. It should be something that people would enjoy reading.
+            It should be safe for work and not contain any inappropriate content.
+            Start with a title for the post and then write the content. Add two new lines after the title.
+            Don't include any quotes or additional text in the output.
+            Don't prefix the title with 'Title:', or content with the word 'Content:'.",
+            "temperature" => 1,
+            "max_tokens" => 256,
+            "top_p" => 1,
+            "frequency_penalty" => 0,
+            "presence_penalty" => 0,
+        ]),
+        CURLOPT_HTTPHEADER => [
+            "Content-Type: application/json",
+            "Authorization: Bearer " . $open_ai_api_key,
+        ],
+    ]);
+    $response = curl_exec($curl);
+    curl_close($curl);
+
+    $response = json_decode($response, true);
+    if (empty($response["choices"])) {
+        response("error", "No response from OpenAI");
+    }
+    try {
+        $output = $response["choices"][0]["text"];
+        response("success", $output);
+    } catch (Exception $e) {
+        response("error", "Invalid response from OpenAI");
+    }
+}
+
+function readImage($image)
+{
+    global $open_ai_api_key;
+    $curl = curl_init();
+    curl_setopt_array($curl, [
+        CURLOPT_URL => "https://api.openai.com/v1/chat/completions",
+        CURLOPT_RETURNTRANSFER => true,
+        CURLOPT_CUSTOMREQUEST => "POST",
+        CURLOPT_POSTFIELDS => json_encode([
+            "model" => "gpt-4-vision-preview",
+            "messages" => [
+                [
+                    "role" => "user",
+                    "content" => [
+                        [
+                            "type" => "text",
+                            "text" => "The image will contain puzzle pieces with text on them.
+                            Please list all the puzzle pieces by only outputting the text on them, each in new line.
+                            Valid lines of text are:
+                            - Site name
+                            - Post
+                            - /wp-admin/
+                            - Multisite
+                            - Omnisend
+                            - WooCommerce
+                            - Google
+                            - Jetpack
+                            - Elementor
+                            - YITH
+                            - Dynamic.ooo
+                            - PersonalizeWP
+                            - JetFormBuilder
+                            - Fastspring
+                            - Cookiebot
+                            - W3 Total Cache
+                            - SiteGround
+                            - Yoast
+                            If the text is invalid, don't return it.
+                            If you do not see puzzle pieces of paper with valid text on them, simply output 'NO'",
```

Comment:
Will the "NO" be seen by humans, or is it a value for an app to interpret? If it will be seen by people, perhaps something like "Unrecognized" would sound more gentle.

Reply:
No, this is internal. We only show a message if a piece was successfully recognized.